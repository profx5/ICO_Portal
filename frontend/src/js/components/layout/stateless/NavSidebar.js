import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {media} from 'js/utils/media';


const NavSidebar = ({children}) => {

    return (
        <React.Fragment>
            <Sidebar>{children}</Sidebar>
        </React.Fragment>
    );
}

NavSidebar.propTypes = {
    children: PropTypes.element
}


export default NavSidebar;

const Sidebar = styled.aside`
    background: white;
    width: 79px;
    min-width: 79px;
    border-right: 1px solid #e6e8f1;
    padding-top: 150px;
    ${media.xs} {
        display: none;
    } 
`;
