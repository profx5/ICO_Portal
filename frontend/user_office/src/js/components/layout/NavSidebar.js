import React from 'react';
import styled from 'styled-components';

import Nav from './components/Nav';


class NavSidebar extends React.Component {

    render() {
        return (

            <Sidebar>
                <Nav />
            </Sidebar>            
        );
    }
}


export default NavSidebar;



// STYLES

const Sidebar = styled.aside`

    background: white;
    width: 79px;
    min-width: 79px;
    border-right: 1px solid #e6e8f1;
    padding-top: 150px;
`;