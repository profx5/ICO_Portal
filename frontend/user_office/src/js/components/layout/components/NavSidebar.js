import React from 'react';
import styled from 'styled-components';

import Nav from './Nav';


const NavSidebar = ({children}) => {

    return (

        <Sidebar>
            {children}
        </Sidebar>            
    );
}


export default NavSidebar;

const Sidebar = styled.aside`

    background: white;
    width: 79px;
    min-width: 79px;
    border-right: 1px solid #e6e8f1;
    padding-top: 150px;
`;
