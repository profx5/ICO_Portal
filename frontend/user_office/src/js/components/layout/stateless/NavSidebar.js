import React from 'react';
import styled from 'styled-components';
import {media} from 'js/utils/media';


const NavSidebar = ({children}) => {

    return (
        <React.Fragment>
            <Sidebar>{children}</Sidebar>
        </React.Fragment>
    );
}


export default NavSidebar;

const Sidebar = styled.aside`
    background: white;
    width: 79px;
    min-width: 79px;
    border-right: 1px solid #e6e8f1;
    padding-top: 150px;
    ${media.xs} {
        display: none;
    } 
`;
