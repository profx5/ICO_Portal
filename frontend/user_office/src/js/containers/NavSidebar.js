import React from 'react';
import {connect} from 'react-redux';

import styled from 'styled-components';

import Nav from '../components/Nav';



class NavSidebar extends React.Component {

    render() {
        return (

            <Sidebar>
                <Nav />
            </Sidebar>            
        );
    }
}


export default NavSidebar;



// STYLES

const Sidebar = styled.aside`

    background: #3172FD;
    width: 105px;
    min-width: 105px;
`;