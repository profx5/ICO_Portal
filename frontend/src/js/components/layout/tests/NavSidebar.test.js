import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import NavSidebar from 'js/components/layout/stateless/NavSidebar';


configure({ adapter: new Adapter() });

it('should render stateless NavSidebar', () => {
    const component = shallow(<NavSidebar/>);
    expect(component).toMatchSnapshot();
})
