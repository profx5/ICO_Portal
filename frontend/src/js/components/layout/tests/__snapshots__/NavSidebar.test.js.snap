// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless NavSidebar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavSidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    background: white;
    width: 79px;
    min-width: 79px;
    border-right: 1px solid #e6e8f1;
    padding-top: 150px;
    ",
            "@media (max-width: 767px)",
            " {
        display: none;
    } 
",
          ],
        },
        "displayName": "styled.aside",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "aside",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
    background: white;
    width: 79px;
    min-width: 79px;
    border-right: 1px solid #e6e8f1;
    padding-top: 150px;
    ",
              "@media (max-width: 767px)",
              " {
        display: none;
    } 
",
            ],
          },
          "displayName": "styled.aside",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "aside",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
